lib/pester.rb

```diff
@@ -80,8 +74,29 @@ def self.retry_action(opts = {}, &block)
 
   private
 
+  def self.should_retry?(e, opts = {})
+    retry_error_classes = opts[:retry_error_classes]
+    retry_error_messages = opts[:retry_error_messages]
+    reraise_error_classes = opts[:reraise_error_classes]
+
+    if retry_error_classes
+      if retry_error_messages
+        retry_error_classes.include?(e.class) && retry_error_messages.any? { |m| e.message[m] }
+      else
+        retry_error_classes.include?(e.class)
+      end
+    elsif retry_error_messages
+      retry_error_messages.any? { |m| e.message[m] }
+    elsif reraise_error_classes && reraise_error_classes.include?(e.class)
+      false
+    else
+      true
+    end
```

Comment:
i would vote for the more verbose but more readable approach

On Wed, Apr 1, 2015 at 2:52 PM, Marc Bollinger <notifications@github.com>
wrote:

> In lib/pester.rb
> <https://github.com/lumoslabs/pester/pull/5#discussion_r27615689>:
>
> > +    retry_error_messages = opts[:retry_error_messages]
> > +    reraise_error_classes = opts[:reraise_error_classes]
> > +
> > +    if retry_error_classes
> > +      if retry_error_messages
> > +        retry_error_classes.include?(e.class) && retry_error_messages.any? { |m| e.message[m] }
> > +      else
> > +        retry_error_classes.include?(e.class)
> > +      end
> > +    elsif retry_error_messages
> > +      retry_error_messages.any? { |m| e.message[m] }
> > +    elsif reraise_error_classes && reraise_error_classes.include?(e.class)
> > +      false
> > +    else
> > +      true
> > +    end
>
> You can technically merge those last to clauses from:
>
> elsif reraise_error_classes && reraise_error_classes.include?(e.class)
>   false
> else
>   true
> end
>
> to:
>
> else
>   !(reraise_error_classes && reraise_error_classes.include?(e.class))
> end
>
> part of me feels like that's more straightforward, part of me feels like
> that's less readable
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lumoslabs/pester/pull/5/files#r27615689>.
>